### PR TITLE
Root Handler

### PR DESCRIPTION
**Pull request by l-monninger**
# Summary
- **RFCs**: $\emptyset$.
- **Categories**:  `networks`
- **Issues**: #236 
- **Associated PRs:** https://github.com/movementlabsxyz/aptos-core/pull/51

Adds `root_handler` and other service paths to `maptos-opt-executor` from the uptream

# Changelog

# Testing

# Outstanding issues

### CODE REVIEW

**Comment by l-monninger** [comment]
gali

**Comment by l-monninger** [comment]
@PrasoonPratham  Here's a dynamic environment running the root: https://github.com/movementlabsxyz/infra/pull/117#issuecomment-2241782454

```
  suzuka-devnet
  aptos-server-name =
  l-monninger.root-handler.aptos.devnet.suzuka.movementlabs.xyz
  aws-ssm-connect-command = aws ssm start-session --region us-east-1 --target
  i-037b04cd425200e99
  faucet-server-name =
  l-monninger.root-handler.faucet.devnet.suzuka.movementlabs.xyz
  m1-da-light-node-server-name =
  l-monninger.root-handler.m1-da-light-node.devnet.suzuka.movementlabs.xyz
  suzuka-node-ec2-name = full-node-a-big-hyena-lm-rh
  suzuka-node-instance-id = i-037b04cd425200e99
  suzuka-node-public-ip = 44.223.77.65
  suzuka-node-region = use1
```

File: protocol-units/execution/opt-executor/src/executor/services.rs [hunk 1]
```diff
@@ -14,15 +15,29 @@ impl Executor {
...
 		let cors = Cors::new()
 			.allow_methods(vec![Method::GET, Method::POST])
 			.allow_credentials(true);
-		let app = Route::new().nest("/v1", api_service).nest("/spec", ui).with(cors);
+		let app = Route::new()
+			.at("/", poem::get(root_handler))
+			.nest("/v1", api_service)
+			.nest("/spec", ui)
+			.at("/spec.json", poem::get(spec_json))
+            .at("/spec.yaml", poem::get(spec_yaml))
```
Comment (mzabaluev) [inline protocol-units/execution/opt-executor/src/executor/services.rs:35]: needs rustfmt

File: protocol-units/execution/opt-executor/src/executor/services.rs [hunk 2]
```diff
@@ -14,15 +15,29 @@ impl Executor {
 	pub async fn run_service(&self) -> Result<(), anyhow::Error> {
 		info!("Starting maptos-opt-executor services at: {:?}", self.listen_url);
 
+		let size_limit = self.context.content_length_limit();
```
Comment (mzabaluev) [inline protocol-units/execution/opt-executor/src/executor/services.rs:18]: This does not seem to be used for anything.

Comment (l-monninger) [reply protocol-units/execution/opt-executor/src/executor/services.rs:18]: Yeah, sorry, it should have been used the same way as in Aptos, but it involves making visibility changes that I didn't have time for. I can remove later.

**Comment by Rahat-ch** [comment]
> @PrasoonPratham Here's a dynamic environment running the root: [movementlabsxyz/infra#117 (comment)](https://github.com/movementlabsxyz/infra/pull/117#issuecomment-2241782454)
> 
> ```
>   suzuka-devnet
>   aptos-server-name =
>   l-monninger.root-handler.aptos.devnet.suzuka.movementlabs.xyz
>   aws-ssm-connect-command = aws ssm start-session --region us-east-1 --target
>   i-037b04cd425200e99
>   faucet-server-name =
>   l-monninger.root-handler.faucet.devnet.suzuka.movementlabs.xyz
>   m1-da-light-node-server-name =
>   l-monninger.root-handler.m1-da-light-node.devnet.suzuka.movementlabs.xyz
>   suzuka-node-ec2-name = full-node-a-big-hyena-lm-rh
>   suzuka-node-instance-id = i-037b04cd425200e99
>   suzuka-node-public-ip = 44.223.77.65
>   suzuka-node-region = use1
> ```
> 
> Can you confirm this is behaving as expected?

Tested this by adding network to petra and funding wallet however issue still persists on petra wallet ui
Screen Shot 2024-07-25 at 4 59 00 PM

File: protocol-units/execution/opt-executor/src/executor/services.rs (hunk 1 above)
Comment (l-monninger) [reply protocol-units/execution/opt-executor/src/executor/services.rs]: Whelp, now this is a `cargo fmt` commit.

**Comment by mzabaluev** [comment]
Please correct the link to the aptos-core PR.

**Review by mzabaluev — COMMENTED** [review]
Looks good besides the spurious file.
Please update the aptos-core commit to the tip of `movement`.

I could do without the global formatting PR (and I took care to review excluding it), just formatting the file with changes. But we'll need to enforce formatting somehow, so feel free to leave this in.

One day we should add a CI check running rustfmt and maybe disable tab formatting so that the diffs looks nicer in GitHub.

**Review by mzabaluev — APPROVED** [review]
Looks good to go. I have no idea why the check workflow has failed, the log looks truncated with no diagnostics.

**Pull request merged**